src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -51,142 +34,177 @@ private ExpressionSyntax ParseInterpolatedStringToken()
+                        if (text.Length > 0)
+                        {
+                            builder.Add(SyntaxFactory.InterpolatedStringText(MakeStringToken(text, text, isVerbatim, SyntaxKind.InterpolatedStringTextToken)));
+                        }
+
+                        builder.Add(ParseInterpolation(this.Options, originalText, interpolation, isVerbatim));
+                    }
+
+                    // Add a token for text following the last interpolation
+                    var lastText = originalText[new Range(interpolations[^1].CloseBraceRange.End, closeQuoteRange.Start)];
+                    if (lastText.Length > 0)
+                    {
+                        var token = MakeStringToken(lastText, lastText, isVerbatim, SyntaxKind.InterpolatedStringTextToken);
+                        builder.Add(SyntaxFactory.InterpolatedStringText(token));
+                    }
                 }
+
+                CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> result = builder;
+                _pool.Free(builder);
+                return result;
             }
 
-            interpolations.Free();
-            var result = SyntaxFactory.InterpolatedStringExpression(openQuote, builder, closeQuote);
-            _pool.Free(builder);
-            if (error != null)
+            SyntaxToken getCloseQuote(Range openQuoteRange)
             {
-                result = result.WithDiagnosticsGreen(new[] { error });
+                // Make a token for the close quote " (even if it was missing)
+                var closeQuoteText = originalText[closeQuoteRange];
+                return closeQuoteText == ""
+                    ? SyntaxFactory.MissingToken(SyntaxKind.InterpolatedStringEndToken).TokenWithTrailingTrivia(originalToken.GetTrailingTrivia())
+                    : SyntaxFactory.Token(null, SyntaxKind.InterpolatedStringEndToken, closeQuoteText, closeQuoteText, originalToken.GetTrailingTrivia());
             }
+        }
 
-            Debug.Assert(originalToken.ToFullString() == result.ToFullString()); // yield from text equals yield from node
-            return CheckFeatureAvailability(result, MessageID.IDS_FeatureInterpolatedStrings);
+        private static InterpolationSyntax ParseInterpolation(CSharpParseOptions options, string text, Lexer.Interpolation interpolation, bool isVerbatim)
+        {
+            // Grab from before the { all the way to the start of the } (or the start of the : if present).  The parsing
+            // of the colon and/or close curly is specially handled in ParseInterpolation below.
+            var parsedText = text[new Range(
+                interpolation.OpenBraceRange.Start,
+                interpolation.HasColon ? interpolation.ColonRange.Start : interpolation.CloseBraceRange.Start)];
+
+            // TODO: some of the trivia in the interpolation maybe should be trailing trivia of the openBraceToken
```

Comment:
note: this todo is still accurate.  trivia is not handled properly with interpolations today.  i fix that in a followup PR>